MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -748,9 +775,62 @@ void EnggDiffractionViewQtGUI::dataCurvesFactory(
   }
 
   m_uiTabFitting.dataPlot->replot();
+  m_zoomTool->setZoomBase();
+  // enable zoom & select peak btn after the plotting on graph
+  setZoomTool(true);
+  m_uiTabFitting.pushButton_select_peak->setEnabled(true);
   data.clear();
 }
 
+void EnggDiffractionViewQtGUI::setPeakPickerEnabled(bool enabled) {
+  m_peakPicker->setEnabled(enabled);
+  m_peakPicker->setVisible(enabled);
+  m_uiTabFitting.dataPlot->replot(); // PeakPicker might get hidden/shown
+  m_uiTabFitting.pushButton_add_peak->setEnabled(enabled);
+  if (enabled) {
+    QString btnText = "Reset Peak Selector";
+    m_uiTabFitting.pushButton_select_peak->setText(btnText);
+  }
+}
+
+void EnggDiffractionViewQtGUI::setPeakPicker(
+    const IPeakFunction_const_sptr &peak) {
+  m_peakPicker->setPeak(peak);
+  m_uiTabFitting.dataPlot->replot();
+}
+
+double EnggDiffractionViewQtGUI::getPeakCentre() const {
+  auto peak = m_peakPicker->peak();
+  auto centre = peak->centre();
+  return centre;
+}
+
+void EnggDiffractionViewQtGUI::fittingWriteFile(const std::string &fileDir) {
+  auto *stream_c_str = fileDir.c_str();
+  auto outfile = std::ofstream(stream_c_str);
```

Comment:
If you remove 809 and replace 810 with something like this:
```
std::ofstream outfile(fileDir.c_str());
```
it should work well. it seems that MSVC is happy with the copy constructor, but gcc complains about it being explicitly removed.